RecoTracker/LSTCore/src/alpaka/PixelTriplet.h

```diff
@@ -0,0 +1,2770 @@
+        acc, alpaka::math::abs(acc, pt_beta), SDL::pt_betaMax);  //need to confirm the range-out value of 7 GeV
+    const float dBetaMuls2 = sdlThetaMulsF2 * 16.f / (min_ptBeta_ptBetaMax * min_ptBeta_ptBetaMax);
+
+    const float alphaInAbsReg =
+        alpaka::math::max(acc,
+                          alpaka::math::abs(acc, alpha_InLo),
+                          alpaka::math::asin(acc, alpaka::math::min(acc, rt_InUp * k2Rinv1GeVf / 3.0f, sinAlphaMax)));
+    const float alphaOutAbsReg =
+        alpaka::math::max(acc,
+                          alpaka::math::abs(acc, alpha_OutLo),
+                          alpaka::math::asin(acc, alpaka::math::min(acc, rt_OutLo * k2Rinv1GeVf / 3.0f, sinAlphaMax)));
+    const float dBetaInLum = lIn < 11 ? 0.0f : alpaka::math::abs(acc, alphaInAbsReg * deltaZLum / z_InUp);
+    const float dBetaOutLum = lOut < 11 ? 0.0f : alpaka::math::abs(acc, alphaOutAbsReg * deltaZLum / z_OutLo);
+    const float dBetaLum2 = (dBetaInLum + dBetaOutLum) * (dBetaInLum + dBetaOutLum);
+
+    const float sinDPhi = alpaka::math::sin(acc, dPhi);
+    const float dBetaRIn2 = 0;  // TODO-RH
+
+    float dBetaROut = 0;
+    if (isEC_lastLayer) {
+      dBetaROut =
+          (alpaka::math::sqrt(acc,
+                              mdsInGPU.anchorHighEdgeX[fourthMDIndex] * mdsInGPU.anchorHighEdgeX[fourthMDIndex] +
+                                  mdsInGPU.anchorHighEdgeY[fourthMDIndex] * mdsInGPU.anchorHighEdgeY[fourthMDIndex]) -
+           alpaka::math::sqrt(acc,
+                              mdsInGPU.anchorLowEdgeX[fourthMDIndex] * mdsInGPU.anchorLowEdgeX[fourthMDIndex] +
+                                  mdsInGPU.anchorLowEdgeY[fourthMDIndex] * mdsInGPU.anchorLowEdgeY[fourthMDIndex])) *
+          sinDPhi / drt_tl_axis;
+    }
+
+    const float dBetaROut2 = dBetaROut * dBetaROut;
+
+    betaOutCut =
+        alpaka::math::asin(
+            acc, alpaka::math::min(acc, drt_tl_axis * k2Rinv1GeVf / ptCut, sinAlphaMax))  //FIXME: need faster version
+        + (0.02f / sdOut_d) + alpaka::math::sqrt(acc, dBetaLum2 + dBetaMuls2);
+
+    //Cut #6: The real beta cut
+    if (alpaka::math::abs(acc, betaOut) >= betaOutCut)
+      return false;
+
+    float drt_InSeg = rt_InUp - rt_InLo;
+
+    const float dBetaRes = 0.02f / alpaka::math::min(acc, sdOut_d, drt_InSeg);
+    const float dBetaCut2 =
+        (dBetaRes * dBetaRes * 2.0f + dBetaMuls2 + dBetaLum2 + dBetaRIn2 + dBetaROut2 +
+         0.25f *
+             (alpaka::math::abs(acc, betaInRHmin - betaInRHmax) + alpaka::math::abs(acc, betaOutRHmin - betaOutRHmax)) *
+             (alpaka::math::abs(acc, betaInRHmin - betaInRHmax) + alpaka::math::abs(acc, betaOutRHmin - betaOutRHmax)));
+    float dBeta = betaIn - betaOut;
+#ifdef CUT_VALUE_DEBUG
+    deltaBetaCut = alpaka::math::sqrt(acc, dBetaCut2);
+#endif
+    return dBeta * dBeta <= dBetaCut2;
+  };
+}  // namespace SDL
+#endif
+
+#ifndef PixelQuintuplet_cuh
+#define PixelQuintuplet_cuh
```

Comment:
Comments from Matti:
> Should the code hereafter be in a separate file?

Maybe @slava77, do you remember why `PixelTriplets` and `PixelQuintuplets` were merged? I think it must have been because of the suggestions we got from a hackathon? Shall we just remove these lines and rename the folder to be more descriptive of what it contains?

Reply:
https://github.com/SegmentLinking/TrackLooper/pull/141

Reply:
Are we ok with a rename and removal of lines that makes this look like a separate file?

Reply:
it would be good to recall why/if the register gains are happening.
Naively, one or two header files should give the same result

Reply:
It seems like calling functions from the same file reduced registers wrt. the case where functions are called from separate files. For pT5s, it is common to call functions that are for pT3s, like https://github.com/SegmentLinking/cmssw/blob/1d9b396766940883638fdcab96123b772accc4a1/RecoTracker/LSTCore/src/alpaka/PixelTriplet.h#L2491 hence the files were joined - but in crude way, just copying `PixelQuintuplets.h` into `PixelTriplets.h`...
I don't know the lower level details for this behavior...

Reply:
back then we had `.cu` files and `PixelQuintuplets` were in a separate .cu. So, a merge was clearly practical. The headers may even have stayed separate.

Now we compile everything (relevant for this split)  in a combined Event.dev.cc. 
So, I think that re-split of the .h should be OK

Reply:
Ok, let's put that in the queue of updates.